lib/kennel/file_cache.rb

```diff
@@ -44,11 +44,14 @@ def load_data
     end
 
     def persist
+      return if @data.nil?
+
       dir = File.dirname(@file)
       FileUtils.mkdir_p(dir) unless File.directory?(dir)
 
       Tempfile.create "kennel-file-cache", dir do |tmp|
         Marshal.dump @data, tmp
+        tmp.flush
```

Comment:
that's the good part I think 🤞

Reply:
... actually `.close` would be better

Reply:
No difference really, it's going to close anyway

Reply:
close is more final than flush, I doubt it makes a difference, but we want to say "ok we are done with this file"